File: app/src/main/java/net/araymond/application/Account.java
package net.araymond.application;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;

public class Account {
    String name;
    double balance;
    ArrayList<Transaction> transactions;

    public Account(String name, double balance) {
        this.name = name;
        this.balance = balance;
        transactions = new ArrayList<Transaction>();
    }

    public void newTransaction(String category, String description, double amount, LocalDate date, LocalTime time) {
        transactions.add(new Transaction(category, description, amount, date, time));
        balance += amount;
    }

    public void removeTransaction(int index) {
        transactions.remove(index);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getBalance() {
        return balance;
    }

    public void setBalance(double balance) {
        this.balance = balance;
    }

    public ArrayList<Transaction> getTransactions() {
        return transactions;
    }
}

File: app/src/main/java/net/araymond/application/Account.kt
package net.araymond.application

import java.io.Serializable
import java.time.LocalDate
import java.time.LocalTime

class Account(var name: String, var balance: Double) : Serializable {
    var transactions: ArrayList<Transaction> = ArrayList()

    fun newTransaction(
        category: String?,
        description: String?,
        amount: Double,
        date: LocalDate?,
        time: LocalTime?
    ) {
        transactions.add(Transaction(category!!, description!!, amount, date!!, time!!, this))
        balance += amount
    }

    fun editTransaction(
        category: String,
        description: String,
        amount: Double,
        date: LocalDate,
        time: LocalTime,
        index: Int
    ) {
        transactions[index].editTransaction(category, description, amount, date, time, this)
    }

    fun removeTransaction(index: Int) {
        transactions.removeAt(index)
    }
}